tests/unit/moduleapi/testrdb.tcl

```diff
@@ -128,5 +128,161 @@ tags "modules" {
                 }
             }
         }
+
+        test {Diskless load swapdb, module can use RedisModuleEvent_ReplAsyncLoad CREATE and COMPLETED events on async_loading} {
+            start_server [list overrides [list loadmodule "$testmodule 2"]] {
+                set replica [srv 0 client]
+                set replica_host [srv 0 host]
+                set replica_port [srv 0 port]
+                set replica_log [srv 0 stdout]
+                start_server [list overrides [list loadmodule "$testmodule 2"]] {
+                    set master [srv 0 client]
+                    set master_host [srv 0 host]
+                    set master_port [srv 0 port]
+
+                    # Set master and replica to use diskless replication on swapdb mode
+                    $master config set repl-diskless-sync yes
+                    $master config set repl-diskless-sync-delay 0
+                    $replica config set repl-diskless-load swapdb
+
+                    # Set replica writable so we can check that a key we manually added is served during replication and disappears afterwards
+                    $replica config set replica-read-only no
+
+                    # replica logs
+                    set loglines [count_log_lines -1]
+
+                    # Initial sync to have matching replids between master and replica
+                    $replica replicaof $master_host $master_port
+
+                    # Let replica finish initial sync with master
+                    wait_for_log_messages -1 {"*MASTER <-> REPLICA sync: Finished with success*"} $loglines 50 100
+
+                    # Set global value on module so we can check if module event callbacks in replica will pick it up
+                    $master testrdb.set.before value1_master
+                    $replica testrdb.set.before value1_replica
+
+                    # Put different data sets on the master and replica
+                    # We need to put large keys on the master since the replica replies to info only once in 2mb
+                    $replica debug populate 2000 slave 10
+                    $master debug populate 1000 master 100000
+                    $master config set rdbcompression no
+
+                    # Force the replica to try another full sync (this time it will have matching master replid)
+                    $master multi
+                    $master client kill type replica
+                    $master set asdf asdf
+                    # the side effect of resizing the backlog is that it is flushed (16k is the min size)
```

Comment:
this will need to be adjusted once #9323 is merged (not sure which one will be merged first)
@ShooterIT FYI